### PR TITLE
client closes on cds shutdown

### LINKED ISSUE #34: close redis connections on cds.shutdown

### PR DESCRIPTION
**Pull request by rlindner81**
Fixes #34

### CODE REVIEW

File: src/redisWrapper.js [hunk 1]
```diff
@@ -128,23 +128,26 @@ const _createClientAndConnect = async (errorHandler) => {
...
     try {
       await client.quit();
     } catch (err) {
-      quitError = err;
-    }
-    if (quitError) {
-      _logErrorOnEvent(
-        new VError({ name: VERROR_CLUSTER_NAME, cause: quitError, info: { clientName } }, "error during client quit")
-      );
+      return err;
```
Comment (soccermax) [inline src/redisWrapper.js:136]: why start such a syntax with quite errors here? Why not simply use the try...catch flow as in most places?

File: src/redisWrapper.js [hunk 2]
```diff
@@ -128,23 +128,26 @@ const _createClientAndConnect = async (errorHandler) => {
...
     }
   }
 };
 
+const _clientErrorHandlerBase = async (client, err, clientName) => {
+  _logErrorOnEvent(new VError({ name: VERROR_CLUSTER_NAME, cause: err, info: { clientName } }, "caught error event"));
+  const quitError = await _closeClientBase(client);
+  if (quitError) {
```
Comment (soccermax) [inline src/redisWrapper.js:144]: see comment above

File: src/redisWrapper.js [hunk 3]
```diff
@@ -165,6 +168,14 @@ const getMainClient = async () => {
   return mainClient;
 };
 
+/**
+ * Closes the main Redis client if it is open. Returns any error that occurs during close or undefined.
+ *
+ * @returns {Error|undefined}
+ * @private
+ */
+const closeMainClient = async () => await _closeClientBase(mainClient);
```
Comment (soccermax) [inline src/redisWrapper.js:177]: would remove base as suffix

**Review by soccermax — APPROVED** [review]
Not sure about that `quite` syntax. But that's up to you

File: src/redisWrapper.js (hunk 1 above)
Comment (rlindner81) [reply src/redisWrapper.js:136]: That's a fair point. I will change that...

File: src/redisWrapper.js (hunk 3 above)
Comment (rlindner81) [reply src/redisWrapper.js:177]: I always use base when it's usable with both main and subscriber client and I like the explicitness.

**Pull request merged**